packages/ra-core/src/dataProvider/useDataProvider.spec.tsx

```diff
@@ -102,16 +102,13 @@ describe('useDataProvider', () => {
             throw new Error('foo');
         });
         const dataProvider = { getOne };
-        const r = () =>
-            render(
-                <CoreAdminContext dataProvider={dataProvider}>
-                    <UseGetOne />
-                </CoreAdminContext>
-            );
-        expect(r).toThrow(
-            new Error(
-                'The dataProvider threw an error. It should return a rejected Promise instead.'
-            )
+        render(
+            <CoreAdminContext dataProvider={dataProvider}>
+                <UseGetOne />
+            </CoreAdminContext>
+        );
+        await screen.findByText(
```

Comment:
I suspect these tests no longer work because the new data router takes one extra tick to actually render the child route. I fixed this test (an similar others) by leveraging the fact that react-admin includes an `ErrorBoundary` in the default layout, which will render any rendering-phase error message on screen.

Reply:
**Nitpick**: You should probably change the test description then